orm/src/main/java/org/hibernate/search/query/hibernate/impl/FullTextQueryImpl.java

```diff
@@ -289,11 +290,20 @@ public FullTextQuery setResultTransformer(ResultTransformer transformer) {
 		return this;
 	}
 
+	/*
+	 * Implementation note: this method is defined as generic in the interface,
+	 * but we must implement it without generics (otherwise it won't compile).
+	 *
+	 * The actual reason is a bit hard to explain: basically we implement
+	 * javax.persistence.Query as a raw type at some point, and our superclass
+	 * (also extended as a raw type) also implements this interface, but as a non-raw type.
+	 * This seems to confuse the compiler, which thinks there are two different methods.
+	 */
```

Comment:
thanks @yrodiere, no need. I'll soon test this and other integrations with a previous build of Infinispan and I'm quite optimistic.

I'll merge this for now so we can move on, but if someone knows I'd be interested.. looks like a Java compiler bug, but then again it would be quite a coincidence that both javac and the Eclipse compiler have the same quirk.